sync/requests.go

```diff
@@ -100,51 +100,51 @@ func TxReqFactory(ids []types.TransactionId) RequestFactory {
 	}
 }
 
-func PoetReqFactory(poetProofRef []byte) RequestFactory {
+func ATxReqFactory(ids []types.AtxId) RequestFactory {
 	return func(s *server.MessageServer, peer p2p.Peer) (chan interface{}, error) {
 		ch := make(chan interface{}, 1)
-		resHandler := func(msg []byte) {
-			s.Info("handle PoET proof response")
+		foo := func(msg []byte) {
+			s.Info("handle atx response ")
 			defer close(ch)
-			var proofMessage types.PoetProofMessage
-			err := types.BytesToInterface(msg, &proofMessage)
+			var tx []types.ActivationTx
+			err := types.BytesToInterface(msg, &tx)
 			if err != nil {
-				s.Error("could not unmarshal PoET proof message: %v", err)
+				s.Error("could not unmarshal tx data %v", err)
 				return
 			}
 
-			ch <- proofMessage
+			ch <- tx
```

Comment:
don't we need to validate that the response matches what was requested?

Reply:
I'll open a ticket for all factories to validate the response with respect to what was requested